tests/Unit/Domain/DomainCreators/Test_Rectangle.cpp

```diff
@@ -0,0 +1,104 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "Domain/CoordinateMaps/AffineMap.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/DomainCreators/Rectangle.hpp"
+#include "Utilities/MakeVector.hpp"
+#include "tests/Unit/Domain/CoordinateMaps/TestMapHelpers.hpp"
+#include "tests/Unit/Domain/TestDomainHelpers.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+using affine_map = CoordinateMaps::AffineMap;
+using affine_map_2d = CoordinateMaps::ProductOf2Maps<affine_map, affine_map>;
+void test_rectangle(const DomainCreators::Rectangle& rectangle,
+                    const std::array<double, 2>& lower_bound,
+                    const std::array<double, 2>& upper_bound,
+                    const std::array<size_t, 2>& expected_extents,
+                    const std::array<size_t, 2>& expected_refinement_level) {
+  const auto domain = rectangle.create_domain();
+  const auto& block = domain.blocks()[0];
+  const auto& neighbors = block.neighbors();
+  const auto& external_boundaries = block.external_boundaries();
+
+  CHECK(block.id() == 0);
+  CHECK(rectangle.initial_extents(0) == expected_extents);
+  CHECK(rectangle.initial_refinement_levels(0) == expected_refinement_level);
+
+  test_domain_construction(
+      domain,
+      std::vector<std::unordered_map<Direction<2>, BlockNeighbor<2>>>{{}},
+      std::vector<std::unordered_set<Direction<2>>>{
+          {{Direction<2>::lower_xi()},
+           {Direction<2>::upper_xi()},
+           {Direction<2>::lower_eta()},
+           {Direction<2>::upper_eta()}}},
+      make_vector(make_coordinate_map<Frame::Logical, Frame::Inertial>(
+          affine_map_2d{affine_map{-1., 1., lower_bound[0], upper_bound[0]},
+                        affine_map{-1., 1., lower_bound[1], upper_bound[1]}})));
+}
+
+void test_periodic_rectangle(
```

Comment:
need to test other periodic boundaries as well (only periodic in x, periodic in x and y, etc)